### PR TITLE
Add support for k8s service accounts

### PR DESCRIPTION
**Pull request by bpblanken**
I'm more or less just following [the steps here](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) + what Steve had mentioned [on the original ticket](https://github.com/broadinstitute/tgg-sre/issues/64)

### CODE REVIEW

**Review by hanars — APPROVED** [review]
looks fine to me!

File: charts/seqr/templates/deployment.yaml [hunk 1]
```diff
@@ -36,6 +36,7 @@ spec:
       imagePullSecrets:
         {{- toYaml . | nindent 6 }}
       {{- end }}
+      serviceAccountName: {{ include "seqr.fullname" . }}
```
Comment (sjahl) [inline charts/seqr/templates/deployment.yaml:39]: i feel like this needs to be gated with logic -- if serviceAccount.create is false, and we try to reference one that doesn't exist, i believe k8s can't create the pod/deployment.

Comment (bpblanken) [reply charts/seqr/templates/deployment.yaml:39]: hrm interesting.  I agree with you, but the helm chart boilerplate didn't gate this line in the same way it did with the service account manifest itself.  I think gating is better though.

Comment (sjahl) [reply charts/seqr/templates/deployment.yaml:39]: ah, the helm chart boilerplate looks like it has a special _helper template for the service account name, which resolves to "default" when there's no service account being created. Which, TIL there's always a service account called default:

```
➜ kubectl get sa
NAME       SECRETS   AGE
default    0         26h
```

**Pull request merged**